src/snowflake/snowpark/session.py

```diff
@@ -541,6 +545,11 @@ def __init__(
                 _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
             )
         )
+        self._use_scoped_temp_read_only_table: bool = (
+            self._conn._get_client_side_session_parameter(
+                _PYTHON_SNOWPARK_ENABLE_SCOPED_TEMP_READ_ONLY_TABLE, False
```

Comment:
If it's for native app, I think it's easier to look up from `self._conn._conn._session_parameters['<param_name>']`, similar to what we did for `ENABLE_FIX_1375538`, WDYT @sfc-gh-sfan?

Reply:
oh, i believe that function does look into the _conn._session_parameters, here is the implementation 
```
    def _get_client_side_session_parameter(self, name: str, default_value: Any) -> Any:
        """It doesn't go to Snowflake to retrieve the session parameter.
        Use this only when you know the Snowflake session parameter is sent to the client when a session/connection is created.
        """
        return (
            self._conn._session_parameters.get(name, default_value)
            if self._conn._session_parameters
            else default_value
        )
```

Reply:
Cool! I think you need to define a server side parameter too, similar to `PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION` above: https://sourcegraph.c2.us-central1.gcp-dev.app.snowflake.com/github.com/snowflakedb/snowflake/-/blob/GlobalServices/src/main/java/com/snowflake/core/parameters/SnowparkEcosystemParameters.java?L774:42

And default it to false.

Reply:
@sfc-gh-deliu i will follow up with you on this, i though you mentioned that i don't need a server side parameter, and you can set this up in the session automatically

Reply:
@sfc-gh-deliu i double checked with @sfc-gh-sfan , we do not need a definition for this parameter at server side. but he suggest we have a parameter control for turning this on for native app. I think we can do that when doing the server side change